### ISSUE TITLE
Document function return values

### ISSUE THREAD

**Issue by mboes** [tweag/rules_haskell#39]
We're decent at documenting what the arguments to functions mean, but we nearly never document their return value when not `None`. This is flagged as a warning by Skylint.

**Comment by mboes** [comment]
Current Skylint output:

```
haskell/c_compile.bzl:19:3-29:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/c_compile.bzl:35:3-45:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/hsc2hs.bzl:14:3-18:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/hsc2hs.bzl:22:3-27:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/path_utils.bzl:5:3-23:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/path_utils.bzl:39:3-51:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/path_utils.bzl:65:3-72:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/path_utils.bzl:83:3-90:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/path_utils.bzl:110:3-121:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/path_utils.bzl:135:3-142:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/path_utils.bzl:148:3-152:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/path_utils.bzl:148:3-152:5: incomplete docstring: parameter 'ctx' not documented [inconsistent-docstring]
haskell/toolchain.bzl:42:3-46:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/toolchain.bzl:146:3-151:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/toolchain.bzl:219:3-224:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/toolchain.bzl:241:3-246:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/toolchain.bzl:291:3-300:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/toolchain.bzl:291:3-300:5: inconsistent docstring: parameter 'interface_files' appears in docstring but not in function signature [inconsistent-docstring]
haskell/toolchain.bzl:356:3-360:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/toolchain.bzl:364:3-370:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/toolchain.bzl:374:3-378:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/toolchain.bzl:382:3-390:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/tools.bzl:4:3-8:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/tools.bzl:15:3-23:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/tools.bzl:28:3-33:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/tools.bzl:41:3-45:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/tools.bzl:51:3-55:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/tools.bzl:61:3-65:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/tools.bzl:69:3-73:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
haskell/tools.bzl:77:3-81:5: incomplete docstring: the return value is not documented (no 'Returns:' section found) [inconsistent-docstring]
```
